### PR TITLE
[Toast3] ground benchmark

### PR DESCRIPTION
**Pull request by nestordemeure**
Adding the benchmarking code for the ground workflow, factorizing some benchmarking functionalities in a dedicated file and updating ground scheduler parameters in a test file.

### CODE REVIEW

File: src/toast/scripts/benchmarking_utilities.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
+# Copyright (c) 2015-2021 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+
+"""
+Functions common to all benchmarking scripts.
```
Comment (tskisner) [inline src/toast/scripts/benchmarking_utilities.py:6]: Great to have these common functions in a separate file, thanks!

**Review by tskisner — APPROVED** [review]
Looks good!  I only pushed a commit that ran format_source.sh will the latest versions of uncrustify and black.  Those will be in the cmbenv stack on cori soon.

File: src/toast/tests/ops_sim_ground.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def test_exec(self):
                     "--site-alt",
                     "{}".format(site.earthloc.height.to_value(u.meter)),
                     "--patch",
-                    "small_patch,1,40,-40,44,-44",
+                    "tile000,1,40,0,60,-40",
```
Comment (keskitalo) [inline src/toast/tests/ops_sim_ground.py:84]: The unit test should stick to the `small_patch` target to keep the running time short.

File: src/toast/unported/tests/ops_dipole.py [hunk 3]
```diff
@@ -149,13 +149,28 @@ def test_sim(self):
         # construct distributed maps to store the covariance,
         # noise weighted map, and hits
 
-        invnpp = DistPixels(self.data, nnz=1, dtype=np.float64, nest=False,)
+        invnpp = DistPixels(
```
Comment (keskitalo) [inline src/toast/unported/tests/ops_dipole.py:152]: `black` seems to have a hard time deciding how to format these lines.

**Review by keskitalo — COMMENTED** [review]
Looks good. I, too, appreciate putting common benchmark code into a separate file.

If you can restore the original patch definition for the unit test, that would make more sense. I think we only need a "realistic" patch definition for benchmarking.

File: src/toast/tests/ops_sim_ground.py (hunk 2 above)
Comment (nestordemeure) [reply src/toast/tests/ops_sim_ground.py:84]: Ok, i will revert it now.

File: src/toast/unported/tests/ops_dipole.py (hunk 3 above)
Comment (nestordemeure) [reply src/toast/unported/tests/ops_dipole.py:152]: Yes, not a fan of what it did to some of the code...

Comment (tskisner) [reply src/toast/unported/tests/ops_dipole.py:152]: The challenge is that black defaults change slightly between versions, and we are all using different versions.  I have just updated the version in the latest cmbenv stack at nersc, and have installed the latest version locally.  It is slightly painful, but I'll just remind everyone the utter formatting hell that we went through in the recent past.  I feel these annoyances are much better than inconsistency everywhere...

**Pull request merged**